File: selection_sort_ts/selection_sort.js
var selectionSort = function (arr) {
  for (var i = 0; i < arr.length; i++) {
    var minIndex = i;
    for (var j = i + 1; j < arr.length; j++) {
      if (arr[minIndex] > arr[j]) {
        minIndex = j;
      }
    }
    var temp = arr[i];
    arr[i] = arr[minIndex];
    arr[minIndex] = temp;
  }
  return arr;
};
var arr = [8, 11, 1, 9, 10, 5, 7, 6, 3, 2, 4];
console.log(selectionSort(arr));

File: selection_sort_java/SelectionSort.java
import java.util.Arrays;

public class SelectionSort {
    public static void main(String[] args) {
        int[] arr = { 5, 1, 9, 3, 10, 2, 6, 4, 8, 7 };
        int[] result = selectionSort(arr);
        System.out.println(Arrays.toString(result));
    }

    public static int[] selectionSort(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            int minIndex = i;
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[minIndex] > arr[j]) {
                    minIndex = j;
                }
            }
            int temp = arr[i];
            arr[i] = arr[minIndex];
            arr[minIndex] = temp;
        }
        return arr;
    }
}
